CodeEdit/Features/Tasks/Shell/TaskShell.swift

```diff
@@ -0,0 +1,74 @@
+//
+//  Shell.swift
+//  CodeEditTests
+//
+//  Created by Tommy Ludwig on 24.06.24.
+//
+
+import Foundation
+
+/// `TaskShell` provides a mechanism for executing shell commands using different shell environments.
+/// It supports a variety of shells including bash, zsh, sh, csh, tcsh, and ksh.
+public enum TaskShell: String {
```

Comment:
Could this instead extend the existing `Shell` enum? It'd mean removing a few of the shells listed here but we'd reduce the duplicate types.